cmd/dusk/server.go

```diff
@@ -299,16 +315,36 @@ func Setup() *Server {
 
 // Close the chain and the connections created through the RPC bus.
 func (s *Server) Close() {
-	// TODO: disconnect peers
-	// _ = s.c.Close(cfg.Get().Database.Driver)
-	s.rpcBus.Close()
-	s.grpcServer.GracefulStop()
+	// Cancel all goroutines long-lived loops
+	s.cancel()
+
+	// Close graphql server.
+	if s.gqlServer != nil {
+		if err := s.gqlServer.Close(); err != nil {
+			log.WithError(err).Warn("failed to close gql server")
+		}
+	}
+
+	if s.grpcServer != nil {
+		s.grpcServer.GracefulStop()
+	}
+
+	// Close Rusk client connection
 	_ = s.ruskConn.Close()
 
-	// kadcast grpc
+	// kadcast client grpc
 	if s.kadPeer != nil {
 		s.kadPeer.Close()
 	}
+
+	if s.dbDriver != nil {
+		if err := s.dbDriver.Close(); err != nil {
+			log.WithError(err).Warn("failed to close db driver")
+		}
+	}
+
+	s.rpcBus.Close()
+	s.eventBus.Close()
```

Comment:
Some of these closures should be done in parallel using a `WaitGroup` to speed up the teardown process